app/src/main/kotlin/com/wire/android/ui/home/sync/FeatureFlagNotificationViewModel.kt

```diff
@@ -72,4 +97,34 @@ class FeatureFlagNotificationViewModel @Inject constructor(
     fun hideDialogStatus() {
         featureFlagState = featureFlagState.copy(showFileSharingDialog = false)
     }
+
+    private suspend fun checkNumberOfSessions(): Int {
+        getSessions().let {
+            return when (it) {
+                is GetAllSessionsResult.Success -> {
+                    it.sessions.filterIsInstance<AccountInfo.Valid>().size
+                }
+
+                is GetAllSessionsResult.Failure.Generic -> 0
+                GetAllSessionsResult.Failure.NoSessionFound -> 0
+            }
+        }
+    }
+
+    fun updateSharingStateIfNeeded(activity: AppCompatActivity) {
```

Comment:
I think this is going to bite our ass if we enable rotation of screen,  ViewModel lives longer then Activity, therefore this is a smell that could memory leak I even think it will cause a leak now if you go to background and back, you can check this out with Canary